packages/gafl-webapp-service/src/pages/summary/licence-summary/__tests__/licence-summary.spec.js

```diff
@@ -107,33 +117,41 @@ describe('The licence summary page', () => {
   })
 
   it('licence length amendment causes a redirect to the summary page', async () => {
+    await injectWithCookies('GET', LICENCE_SUMMARY.uri)
+    const data = await injectWithCookies('GET', LICENCE_LENGTH.uri)
+    // Tests that the back-link is now set to the summary page
+    expect(data.payload.search(backLinkRegEx(LICENCE_SUMMARY.uri)) > 0).toBeTruthy()
     await postRedirectGet(LICENCE_LENGTH.uri, { 'licence-length': '8D' })
-    const data = await injectWithCookies('GET', LICENCE_SUMMARY.uri)
-    expect(data.statusCode).toBe(200)
+    const data2 = await injectWithCookies('GET', LICENCE_SUMMARY.uri)
+    expect(data2.statusCode).toBe(200)
   })
 
-  it('licence type amendments cause an eventual redirect back to the summary page', async () => {
+  it('licence type amendments cause an eventual redirect back to the summary page (trout and coarse)', async () => {
+    await injectWithCookies('GET', LICENCE_SUMMARY.uri)
     await postRedirectGet(LICENCE_LENGTH.uri, { 'licence-length': '12M' })
+    const data = await injectWithCookies('GET', LICENCE_TYPE.uri)
+    expect(data.payload.search(backLinkRegEx(LICENCE_SUMMARY.uri)) > 0).toBeTruthy()
     await postRedirectGet(LICENCE_TYPE.uri, { 'licence-type': 'salmon-and-sea-trout' })
-    const data = await injectWithCookies('GET', LICENCE_SUMMARY.uri)
-    expect(data.statusCode).toBe(200)
-
-    const data2 = await postRedirectGet(LICENCE_TYPE.uri, { 'licence-type': 'trout-and-coarse' })
-
-    expect(data2.statusCode).toBe(302)
-    expect(data2.headers.location).toBe(NUMBER_OF_RODS.uri)
-
-    const data3 = await postRedirectGet(NUMBER_OF_RODS.uri, { 'number-of-rods': '2' })
+    const data2 = await injectWithCookies('GET', LICENCE_SUMMARY.uri)
+    expect(data2.statusCode).toBe(200)
+    const data3 = await postRedirectGet(LICENCE_TYPE.uri, { 'licence-type': 'trout-and-coarse' })
     expect(data3.statusCode).toBe(302)
-    expect(data3.headers.location).toBe(LICENCE_SUMMARY.uri)
+    expect(data3.headers.location).toBe(NUMBER_OF_RODS.uri)
+    const data4 = await injectWithCookies('GET', NUMBER_OF_RODS.uri)
+    expect(data4.payload.search(backLinkRegEx(LICENCE_SUMMARY.uri)) > 0).toBeTruthy()
+    const data5 = await postRedirectGet(NUMBER_OF_RODS.uri, { 'number-of-rods': '2' })
+    expect(data5.statusCode).toBe(302)
+    expect(data5.headers.location).toBe(LICENCE_SUMMARY.uri)
```

Comment:
This test is starting to feel really complex - I can see that you're trying to test an eventual redirect, but there's quite a lot of other stuff being tested too, might be better to break these out to other tests?

Reply:
Simplified